seis/sl/conn.go

```diff
@@ -0,0 +1,389 @@
+	}
+
+	return c.infoCommand(info.command)
+}
+
+// GetInfo requests the seedlink server return an INFO request for the given level. The results
+// are returned as a decoded Info pointer, or an error otherwise.
+func (c *Conn) GetInfo(level string) (*Info, error) {
+	data, err := c.GetInfoLevel(level)
+	if err != nil {
+		return nil, err
+	}
+
+	var info Info
+	if err := info.Unmarshal(data); err != nil {
+		return nil, err
+	}
+
+	return &info, nil
+}
+
+// CommandId sends an INFO ID command to the seedlink server.
+func (c *Conn) CommandId() ([]byte, error) {
+	return c.infoCommand(cmdInfoId)
+}
+
+// CommandHello sends a HELLO command to the seedlink server.
+func (c *Conn) CommandHello() ([]byte, error) {
+	return c.infoCommand(cmdHello)
+}
+
+// CommandClose sends a BYE command to the seedlink server.
+func (c *Conn) CommandClose() ([]byte, error) {
+	return c.infoCommand(cmdClose)
+}
+
+// CommandStationList sends a CAT command to the seedlink server.
+func (c *Conn) CommandCat() ([]byte, error) {
+	return c.infoCommand(cmdCat)
+}
+
+// CommandStation sends a STATION command to the seedlink server.
+func (c *Conn) CommandStation(station, network string) error {
+	if strings.ContainsAny(station, "*?") && !c.capabilities["NSWILDCARD"] {
+		return fmt.Errorf("station selector '%s' contains wildcards but the server does not report capability NSWILDCARD", station)
+	}
+	if strings.ContainsAny(network, "*?") && !c.capabilities["NSWILDCARD"] {
+		return fmt.Errorf("network selector '%s' contains wildcards but the server does not report capability NSWILDCARD", network)
+	}
+	switch {
+	case network != "":
+		if err := c.modifierCommand(fmt.Sprintf("%s %s %s", cmdStation, station, network)); err != nil {
+			return fmt.Errorf("error sending STATION %s %s: %v", station, network, err)
+		}
+	default:
+		if err := c.modifierCommand(fmt.Sprintf("%s %s", cmdStation, station)); err != nil {
+			return fmt.Errorf("error sending STATION %s: %v", station, err)
+		}
+	}
+
+	return nil
+}
+
+// CommandSelect sends a SELECT command to the seedlink server.
+func (c *Conn) CommandSelect(selection string) error {
+
+	if err := c.modifierCommand(fmt.Sprintf("%s %s", cmdSelect, selection)); err != nil {
+		return fmt.Errorf("error sending SELECT %s: %v", selection, err)
+	}
+
+	return nil
+}
+
+// CommandData sends a DATA command to the seedlink server.
+func (c *Conn) CommandData(sequence string, starttime time.Time) error {
+
+	var dc string
+	switch {
+	case sequence == "":
+		dc = cmdData
+	case starttime.IsZero():
+		dc = fmt.Sprintf("%s %s\n", cmdData, sequence)
+	default:
+		dc = fmt.Sprintf("%s %s %s\n", cmdData, sequence, starttime.Format(timeFormat))
+	}
+
+	if err := c.modifierCommand(dc); err != nil {
+		return fmt.Errorf("error sending DATA: %v", err)
+	}
+
+	return nil
+}
+
+// CommandTime sends a TIME command to the seedlink server.
+func (c *Conn) CommandTime(starttime, endtime time.Time) error {
+
+	if starttime.IsZero() {
+		return nil
```

Comment:
is it intended to do nothing in this case?

Reply:
Yes, don't want to set the time if there isn't one.